### PR TITLE
fix #780 - Update do.sh

### LINKED ISSUE #780: khmer user doc, do.sh script uses the old argument --savehash
In the Khmer user documentation - "A few examples" section: The do.sh uses the old argument --savehash, this needs to be updated to --savetable

### PR DESCRIPTION
**Pull request by drtamermansour**
Fix #780 :In the do.sh script, the old argument --savehash has been updated to --savetable

### CODE REVIEW

**Comment by mr-c** [comment]
Jenkins, add to testlist

**Comment by mr-c** [comment]
Hey, @drtamermansour can you add the line `bash -ex cd examples/stamps/; ./do.sh` to `jenkins-build.sh` as part of this pull request?

**Comment by mr-c** [comment]
You'll also need to change the mode of the 'examples/stamps/do.sh' file. This can only be done at the command line (via `mode u+x examples/stamps/do.sh`)

**Comment by drtamermansour** [comment]
I can not see a special format of the ```jenkins-build.sh file so I just added the line of code to the end of the file. Also I assume that you meant chmod

**Comment by mr-c** [comment]
Jenkins, test this please

File: jenkins-build.sh [hunk 1]
```diff
@@ -64,3 +64,5 @@ if type sloccount >/dev/null 2>&1
 then
 	make sloccount.sc
 fi
+
+bash -ex 'cd examples/stamps/; ./do.sh'
```
Comment (mr-c) [inline jenkins-build.sh]: mea culpa; add ` -c ` after `-ex`

**Comment by mr-c** [comment]
@drtamermansour can you paste in the checklist and fill it out? http://khmer.readthedocs.org/en/latest/dev/coding-guidelines-and-review.html#checklist

**Comment by mr-c** [comment]
Jenkins, test this please

**Comment by drtamermansour** [comment]
- [ ] Is it mergable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage.
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

**Comment by drtamermansour** [comment]
When I try:  make doc test pep8

This is what I get:
./setup.py build_sphinx --fresh-env
/mnt/home/mansourt/env/lib/python2.7/site-packages/setuptools/dist.py:291: UserW
arning: The version specified ('1.3-204-gd89c575-dirty') is an invalid version,
this may not work as expected with newer versions of setuptools, pip, and PyPI.
Please see PEP 440 for more details.
  "details." % self.metadata.version
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: invalid command 'build_sphinx'


**Note**: I am working on MSU-HPC using the virtual environment. I installed the dependencies and the make test passed ok

**Comment by mr-c** [comment]
Turns out that the HPCC used an old version of virtualenv; local upgrade fixed that.

Next he ran into https://github.com/warner/python-versioneer/issues/52 which he fixed via multiple `pip uninstall screed`s followed by `pip install bz2file; pip install -i https://testpypi.python.org/pypi --pre screed`

**Comment by drtamermansour** [comment]
ready for review

**Comment by drtamermansour** [comment]
- [ ] Is it mergable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage.
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

File: jenkins-build.sh [hunk 2]
```diff
@@ -64,3 +64,5 @@ if type sloccount >/dev/null 2>&1
 then
 	make sloccount.sc
 fi
+
+bash -ex -c 'cd examples/stamps/; ./do.sh'
```
Comment (mr-c) [inline jenkins-build.sh]: Oh, I just had an idea: if this test doesn't work there won't be a good indication of why

if you add `|| { echo examples/stamps/do.sh no longer runs; /bin/false }` to the end of line 68 that will fix this.

File: ChangeLog [hunk 3]
```diff
@@ -1648,3 +1653,4 @@ project to conform to the standard layout
 * It is now easy to calculate code coverage.
 
 * setup.py is now PEP8 compliant
+
```
Comment (mr-c) [inline ChangeLog]: Remove this line

**Comment by drtamermansour** [comment]
ready for review

File: ChangeLog [hunk 4]
```diff
@@ -1,3 +1,8 @@
+2015-02-20  Tamer A. Mansour  <drtamermansour@gmail.com>
+
+   * In the do.sh script, the old argument --savehash has been updated to
+--savetable
```
Comment (mr-c) [inline ChangeLog]: Wrapped lines should be indented to match the line above. In this instance you need three space before --savetable

**Comment by mr-c** [comment]
Otherwise this is ready to merge

**Comment by drtamermansour** [comment]
@mr-c  
I do not know why Jenkins failed !!
When I run: make pep8, it gives me 
pep8 --exclude=_version.py setup.py khmer/ scripts/ tests/ || true
Is this an error?

**Comment by ctb** [comment]
This passes 'make test' and 'make pep8' on my laptop; looks like Jenkins may be barfing on it.

Jenkins, test this please.

**Comment by ctb** [comment]
please test this

**Comment by drtamermansour** [comment]
I think you need to mention "Jenkins" in the test request

**Comment by mr-c** [comment]
`jenkins-build.sh: line 69: syntax error: unexpected end of file`

**Comment by drtamermansour** [comment]
Jenkins, test this please

**Comment by mr-c** [comment]
Jenkins, test this please

**Comment by mr-c** [comment]
Thanks, @drtamermansour

**Pull request merged**